prov/efa/src/efa_fabric.c

```diff
@@ -787,15 +788,17 @@ static int efa_set_fi_address(const char *node, const char *service, uint64_t fl
 	}
 
 	if (dest_addr) {
-		fi->dest_addr = malloc(EFA_EP_ADDR_LEN);
-		if (!fi->dest_addr)
-			return -FI_ENOMEM;
-		memcpy(fi->dest_addr, dest_addr, EFA_EP_ADDR_LEN);
+		for (cur = fi; cur; cur = cur->next) {
+			cur->dest_addr = malloc(EFA_EP_ADDR_LEN);
+			if (!cur->dest_addr) {
+				for (; fi->dest_addr; fi = fi->next)
+					free(fi->dest_addr);
```

Comment:
The happy path is relying on a call to fi_freeinfo() elsewhere but the error path is free'ing it explicitly here.  If we're allocating for `EFA_EP_ADDR_LEN`, maybe move the allocation into efa_alloc_info?